src/wasm-interpreter.h

```diff
@@ -1394,7 +1430,10 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     if (parent.breaking()) {
       return parent;
     }
-    return Literal(curr->type);
+    auto parentValue = parent.getSingleValue();
+    RttSupers newSupers = parentValue.getRttSupers();
+    newSupers.push_back(parentValue.type);
```

Comment:
The idea is that `rttSupers` contains just the "parents":

* `rtt.canon` will have an empty `rttSupers`
* `rtt.sub` on that will have one item in `rttSupers`, just the type of the original canon
* etc.

So each time we take the supers from the parent, and add the parent, to get our supers.

Reply:
If we only contain the parents, would the current `isSubRtt` work? For example,
```
Rtt A: 
Parents: a->b
Current: c

Rtt B:
Parents: a->b
Current: d
```
The current implementation seems to return `A <: B` because it only compares A's supers with B's type. I'm not sure if that implementation itself is correct (which I described above), but anyway I think including the current in the chain might be more intuitive, because self is also a supertype of the self.

Reply:
I don't think `A.isSubRtt(B)` is true? We compare a, b, and c (all of A's supers, and also A's actual type right before the loop) to d (B's actual type), and find that none of them are equal to d. So we return false.

Reply:
Sorry my bad. The correct example:
```
Rtt A: 
Parents: a->b
Current: c

Rtt B:
Parents: a->b
Current: a
```

Reply:
I think the current code can correctly reject this case.